org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -416,86 +451,141 @@ private void terminateProcess(IConsumerProcess<U> process) throws EventException
 	public void run() throws EventException {
 		
 		startJobManager();
+		init();
+		
+		while(isActive()) {
+        	try {
+	            boolean ok = consume();
+	            if (!ok) break;
+	                     		
+        	} catch (Throwable ne) {
+        		boolean stayAlive = processException(ne);
+        		if (stayAlive) {
+        			continue;
+        		} else {
+        			break;
+        		}
+        	}
+
+		}
+	}
+
+	private void init() throws EventException {
+		
+		this.waitTime = 0;
 
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS135)